### DISCUSSION TITLE
如何获取文章阅读量

### DISCUSSION THREAD

**Discussion by Konfido** [walinejs/waline#282] [Q&A] [answered]
描述：参考 [doc](https://waline.js.org/client/visitor.html#%E6%96%87%E7%AB%A0%E9%98%85%E8%AF%BB%E9%87%8F%E7%BB%9F%E8%AE%A1) 更新代码后，能获取评论数量，但无法获取阅读量，Leancloud里也没有类似Counter的数据产生。

问题：
- 请问只需要添加doc里展示的代码就可以，还是说Counter的生成/读取/增加/保存 等逻辑需要自己写？
- 能否提供更详细的操作说明

**Comment by lizheming** [comment]
你在 LeanCloud 那边手工创建一个 Counter 表试试？<kbd>数据存储</kbd>-<kbd>结构化数据</kbd>-<kbd>创建Class</kbd>，名称输入 Counter 选择创建即可。

**Reply by Konfido** [reply]
依然不行，我按valine的格式手动添加了一条记录（包括time,title,url）也无法读取展示

**Comment by Mister-Hope** [comment, marked as answer]
你需要保证页面有要求的dom元素结构，同时visitor设置为true。欢迎提供更多信息如果你仍无法使用阅读量。

**Reply by Konfido** [reply]
谢谢，我又看了下自己的theme代码，按下面这样稍微修改后解决了问题。
```diff
// waline.ejs
-   var visitor = '<%= theme.waline.visitor %>' == true ? true : false;

new Waline({
    ...
-   visitor = visitor,
+   visitor = '<%= theme.waline.visitor %>',
    ...
})
```

但现在有另外一个问题，使用valine的 `.leancloud_visitors` 时可以获取阅读量，但按最新doc的指引将其改为waline的样式时，页面上无法显示数字，但leancloud数据库里counter的计数却随着页面的刷新而增加。

```diff
// page_views.ejs
-   <span id="<%- url_for(page.path) %>" class="leancloud_visitors">
-        <span class="meta-seperator">|</span>
-        <i class="fa fa-eye"></i>
-        <i class="leancloud-visitors-count"></i>
-    </span>
+   <span id="<%- url_for(page.path) %>" class="waline-comment-count" />
```

具体代码在 [这里](https://github.com/Konfido/konfido.github.io/blob/hexo/themes/cactus/layout/_partial/post/page_views.ejs#L2-L6)

**Reply by Mister-Hope** [reply]
最新doc是对齐v1的，应该是明天发。

**Reply by Konfido** [reply]
哦哦，了解 👌
